node/src/main/java/io/airlift/node/NodeConfig.java

```diff
@@ -46,6 +46,7 @@ public class NodeConfig
     private String configSpec;
     private AddressSource internalAddressSource = AddressSource.IP;
     private String annotationFile;
+    private boolean preferIpv6Address = "true".equalsIgnoreCase(System.getenv("java.net.preferIPv6Address"));
```

Comment:
@jalpan-randeri shouldn't this be checking `System.getProperty()`, not `System.getenv()`?